contracts/eosio.system/src/producer_pay.cpp

```diff
@@ -88,6 +88,8 @@ namespace eosiosystem {
          auto to_voters        = 2 * to_per_block_pay;
          auto to_savings       = distribute_tokens - (to_voters + to_per_block_pay);
 
+         auto to_producers_pay = to_per_block_pay * 21 / ( _gstate4.standby_pay_ratio * _gstate4.num_standby_slots + 21 );
```

Comment:
we should make this all integer involved in the calculation and the form should probably appear as a NUMERATOR and a DENOMINATOR. This will prevent numerical errors like rounding to zero when we would not want them, and increase precision in the final value. I.e.
to_producers_pay = NUMERATOR/DENOMINATOR
= to_per_block_pay * 21 * STANDBY_PAY_RATIO_DENOMINATOR / (_gstate4.standby_pay_ratio_numerator * _gstate4.num_standby_slots + 21 * STANDBY_PAY_RATIO_DENOMINATOR)